### PR TITLE
Multiple pinned users

### PR DESCRIPTION
**Pull request by pkrucz00**

### CODE REVIEW

File: assets/src/features/room-page/utils/usePinning.tsx [hunk 1]
```diff
@@ -1,15 +1,26 @@
...
+  unpin: (tileId: string) => void;
 };
 
 const usePinning = (): PinningApi => {
-  const [pinnedTileId, setPinnedTileId] = useState<string>("");
+  const [pinnedTileIds, setPinnedTileIds] = useState<string[]>([]);
 
-  return { pinnedTileId, pin: setPinnedTileId, unpin: () => setPinnedTileId("") };
+  const pin = (newTileId: string) => {
+    console.log(pinnedTileIds);
```
Comment (kamil-stasiak) [inline assets/src/features/room-page/utils/usePinning.tsx:13]: pleas remove this console.log

File: assets/src/features/room-page/utils/usePinning.tsx [hunk 2]
```diff
@@ -1,15 +1,26 @@
...
+    if (!pinnedTileIds.includes(newTileId)) setPinnedTileIds((oldPinnedTileIds) => [newTileId, ...oldPinnedTileIds]);
+    console.log(pinnedTileIds);
```
Comment (kamil-stasiak) [inline assets/src/features/room-page/utils/usePinning.tsx:15]: pleas remove this console.log

File: assets/src/features/room-page/utils/usePinning.tsx [hunk 3]
```diff
@@ -1,15 +1,26 @@
...
+  };
+
+  const unpin = (tileIdToRemove: string) => {
+    const idsWithoutGivenTile = pinnedTileIds.filter((tileId) => tileId !== tileIdToRemove);
+    setPinnedTileIds(idsWithoutGivenTile);
```
Comment (kamil-stasiak) [inline assets/src/features/room-page/utils/usePinning.tsx:20]: It's safer to make this calculation inside `setPinnedTileIds`. There is most recent state

```ts
    setPinnedTileIds((prevState) => prevState.filter((tileId) => tileId !== tileIdToRemove));
```

File: assets/src/pages/room/components/StreamPlayer/PinnedTilesSection.tsx [hunk 4]
```diff
@@ -11,48 +11,58 @@ import {
...
+              video={pinnedTile.video}
+              audio={tileType === "remote" ? pinnedTile.audio : null}
+              streamSource={pinnedTile.streamSource}
+              blockFillContent={tileType === "screenShare"}
+              flipHorizontally={tileType === "local"}
+              layers={
+                <>
+                  {hasInitials && showDisabledIcon(pinnedTile.video) ? (
+                    <InitialsImage initials={pinnedTile.initials} />
+                  ) : null}
```
Comment (kamil-stasiak) [inline assets/src/pages/room/components/StreamPlayer/PinnedTilesSection.tsx:47]: Shluld work without elivs operator:

```{hasInitials && showDisabledIcon(pinnedTile.video) && <InitialsImage initials={pinnedTile.initials} />}```

File: assets/src/pages/room/components/StreamPlayer/PinnedTilesSection.tsx [hunk 5]
```diff
@@ -11,48 +11,58 @@ import {
...
+                  <PeerInfoLayer
+                    topRight={<PinIndicator />}
+                    topLeft={
+                      hasInitials && showDisabledIcon(pinnedTile.audio) ? (
+                        <DisabledMicIcon isLoading={isLoading(pinnedTile.audio)} />
+                      ) : null
+                    }
+                    bottomLeft={<NameTag name={pinnedTile.displayName} />}
+                  />
+                  <PinTileButton pinned={true} onClick={() => unpin(pinnedTile.mediaPlayerId)} />
```
Comment (kamil-stasiak) [inline assets/src/pages/room/components/StreamPlayer/PinnedTilesSection.tsx:57]: What do you think about renaming it to "PinTileLayer" or "HoverLayer"?

**Review by kamil-stasiak — CHANGES_REQUESTED** [review]

File: assets/src/pages/room/components/StreamPlayer/PinnedTilesSection.tsx (hunk 5 above)
Comment (pkrucz00) [reply assets/src/pages/room/components/StreamPlayer/PinnedTilesSection.tsx:57]: `PinTileLayer` is fine. I'll rename it

**Review by kamil-stasiak — APPROVED** [review]

**Pull request merged**